### PR TITLE
Add a section about nodelocaldns memory limits.

### PR DESCRIPTION
**Pull request by prameshj**
This was requested in https://github.com/kubernetes/dns/issues/438

### CODE REVIEW

**Comment by prameshj** [comment]
@chrisohaver can you check if this info is correct? Thanks!

**Review by sftim — COMMENTED** [review]
Thanks @prameshj 

Here's some feedback; it's a mixture of wording and content suggestions. Do you think these improve the page?

**Review by chrisohaver — APPROVED** [review]
The CoreDNS specific info lgtm.

File: content/en/docs/tasks/administer-cluster/nodelocaldns.md [hunk 1]
```diff
@@ -100,4 +100,16 @@ shown in [the example](/docs/tasks/administer-cluster/dns-custom-nameservers/#ex
...
+node-local-dns pods use memory for storing cache entries and processing queries. Since they do not watch Kubernetes objects, the cluster size or the number of Services/Endpoints do not affect memory usage. Memory usage is influenced by the DNS query pattern.
+From [CoreDNS docs](https://github.com/coredns/deployment/blob/master/kubernetes/Scaling_CoreDNS.md),
+`The default cache size is 10000 entries, which uses about 30 MB when completely filled.`
+
+This would be the memory usage for each server block (if the cache gets completely filled).
+Memory usage can be reduced by specifying smaller cache sizes.
+
+The number of concurrent queries can lead to additional memory usage (more goroutines). An upper limit can be set via the "max_concurrent" option in the forward plugin.
+
+If a node-local-dns pod gets OOMKilled, it will not cleanup the custom iptables rules added at startup time. The node-local-dns pod should get restarted(since it is part of a daemonset), but this will lead to a brief DNS downtime everytime the pod crashes. A suitable memory limit can be determined by running node-local-dns pods without a limit and measuring the peak usage.
```
[the remark this answers is not in the document]
Comment (prameshj) [reply content/en/docs/tasks/administer-cluster/nodelocaldns.md:115]: wow, this is much better. Thanks!

**Comment by sftim** [comment]
With https://github.com/kubernetes/website/pull/31571#pullrequestreview-867972610 in mind @prameshj, would you be willing to squash commits for this PR?
/lgtm

**Comment by prameshj** [comment]
> With [#31571 (review)](https://github.com/kubernetes/website/pull/31571#pullrequestreview-867972610) in mind @prameshj, would you be willing to squash commits for this PR? /lgtm

Thanks! Done :)

**Comment by sftim** [comment]
Taking LGTM also from https://github.com/kubernetes/website/pull/31571#pullrequestreview-867972610
/approve

**Pull request merged**